web_src/js/components/DiffFileTree.vue

```diff
@@ -113,8 +113,12 @@ export default {
       this.updateState(this.fileTreeIsVisible);
     },
     updateState(visible) {
-      const [toShow, toHide] = document.querySelectorAll('.diff-toggle-file-tree-button .icon');
+      const btn = document.querySelector('.diff-toggle-file-tree-button');
+      const [toShow, toHide] = btn.querySelectorAll('.icon');
       const tree = document.getElementById('diff-file-tree');
+      const newTooltip = btn.getAttribute(`data-content-${visible ? 'hide' : 'show'}`);
+      btn.setAttribute('data-content', newTooltip);
+      btn._tippy.setContent(newTooltip);
```

Comment:
I don't think this code is safe. 

There is no init order constraint, there might be no `btn._tippy` because the "tooltip" module might not be inited?

Reply:
It does run on component mount, so there is a risk. I'll see to not trying to update at mount as the tooltip will already be correct there from the HTML script.

Reply:
How about https://github.com/go-gitea/gitea/pull/23574 ?  With that PR, developers only need to change "data-tooltip-content", they do not need to call `btn._tippy.setContent` manually.

Reply:
Updated it so it skips the tooltip update on mount.

https://github.com/go-gitea/gitea/pull/23574 with the MutationObserver does sound good, but I can't leave out this update now, so we need to adjust PRs depending on which one lands first.

Reply:
https://github.com/go-gitea/gitea/pull/23574 is merged, so let's see if I can skip the `setContent` now.

Reply:
Does not appear to work. The `data-content` attribute is updated, but the tooltip does not change.

Reply:
Yeah, works now after update to `data-tooltip-content`.